sources/address.c

```diff
@@ -28,28 +26,31 @@ int od_address_range_copy(od_address_range_t *src, od_address_range_t *dst)
 	dst->is_hostname = src->is_hostname;
 }
 
-int od_address_range_read_prefix(od_address_range_t *address_range, char *prefix)
+int od_address_range_read_prefix(od_address_range_t *address_range,
+				 char *prefix)
 {
 	char *end = NULL;
-	long len = strtoul(prefix, &end, 10);
+	long len = strtol(prefix, &end, 10);
 	if (*prefix == '\0' || *end != '\0') {
 		return -1;
 	}
 	if (address_range->addr.ss_family == AF_INET) {
 		if (len > 32)
 			return -1;
-		struct sockaddr_in *addr = (struct sockaddr_in *)&address_range->mask;
-		long mask;
+		struct sockaddr_in *addr =
+			(struct sockaddr_in *)&address_range->mask;
+		uint32 mask;
 		if (len > 0)
 			mask = (0xffffffffUL << (32 - (int)len)) & 0xffffffffUL;
```

Comment:
it seems that you now don't need to do "& 0xffffffffUL" here

Reply:
thanks, fixed